test/src/menuui/test_intel_parse.cpp

```diff
@@ -0,0 +1,307 @@
+TEST_F(IntelParseTest, empty_file) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// The same with only white space.
+TEST_F(IntelParseTest, only_white_space) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// A single valid entry.
+TEST_F(IntelParseTest, single) {
+	SCOPED_TRACE("single");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+}
+
+// The same, but with translation to German.
+TEST_F(IntelParseTest, single_translate) {
+	SCOPED_TRACE("single_translate");
+
+	lcl_xstr_close();
+	lcl_set_language(LCL_GERMAN);
+	lcl_xstr_init();
+
+	techroom_intel_init();
+
+	intel_data expected = {
+		"Foo name German", // XSTR id 3000
+		"Foo desc German", // XSTR id 3001
+		"Foo anim",
+		IIF_IN_TECH_DATABASE | IIF_DEFAULT_IN_TECH_DATABASE
+	};
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected, Intel_info[0]);
+}
+
+// Rubbish followed by a valid entry.
+// Parsing stops at the rubbish.
+TEST_F(IntelParseTest, invalid_start) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// A valid entry followed by rubbish on the same line, followed by
+// another valid entry. Parsing stops at the rubbish.
+TEST_F(IntelParseTest, invalid_end_same_line) {
+	SCOPED_TRACE("invalid_end_same_line");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+}
+
+// A valid entry followed by rubbish on a new line, followed by another
+// valid entry. Parsing stops at the rubbish.
+TEST_F(IntelParseTest, invalid_end_new_line) {
+	SCOPED_TRACE("invalid_end_new_line");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+}
+
+// Three valid entries.
+TEST_F(IntelParseTest, three) {
+	SCOPED_TRACE("three");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 3);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+	test_intel_data_equal(expected_bar, Intel_info[1]);
+	test_intel_data_equal(expected_baz, Intel_info[2]);
+}
+
+// Two identical entries are allowed.
+TEST_F(IntelParseTest, identical) {
+	SCOPED_TRACE("identical");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 2);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+	test_intel_data_equal(expected_foo, Intel_info[1]);
+}
+
+// A valid entry followed by one with a missing $:Entry line.
+// The second entry should be ignored. Parsing should stop there.
+TEST_F(IntelParseTest, missing_entry) {
+	SCOPED_TRACE("missing_entry");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+}
+
+// A valid entry followed by one with a missing $Name: line.
+// The second entry should be ignored. Parsing should stop there.
+TEST_F(IntelParseTest, missing_name) {
+	SCOPED_TRACE("missing_name");
+
+	// this might throw, but we don't want to make that contractual
```

Comment:
You should write the test for the code that is used at the moment. If that code may throw exceptions then that should also be tested. If it may not throw exceptions then a thrown exception will make the test fail so you could just remove the try-catch,

Anyway, an exception in a test should never be ignored so you should change the test accordingly.